src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -2578,6 +2578,48 @@ PyObject *cfl_int32_array_to_py_list (const int32_t *arr, size_t cnt) {
 }
 
 
+/****************************************************************************
+ *
+ *
+ * Methods common across all types of clients.
+ *
+ *
+ *
+ *
+ ****************************************************************************/
+
+const char set_sasl_credentials_doc[] = PyDoc_STR(
+        ".. py:function:: set_sasl_credentials(username, password)\n"
+        "\n"
+        "  Sets the SASL credentials used for this client.\n"
+        "  These credentials will overwrite the old ones, and will be used the next time the client needs to authenticate.\n"
+        "  This method will not disconnect existing broker connections that have been established with the old credentials.\n"
+        "  This method is applicable only to SASL PLAIN and SCRAM mechanisms.\n");
+
+
+PyObject *set_sasl_credentials(Handle *self, PyObject *args, PyObject *kwargs) {
+        const char *username = NULL;
+        const char *password = NULL;
+        rd_kafka_error_t* error;
+        static char *kws[] = {"username", "password", NULL};
+
+        if (!PyArg_ParseTupleAndKeywords(args, kwargs, "ss", kws,
+                                         &username, &password)) {
+                return NULL;
+        }
+
+        error = rd_kafka_sasl_set_credentials(self->rk, username, password);
```

Comment:
Check if we need to release GIL.

Reply:
We are actually doing some operations except just copying of credentials
I'm not sure how time consuming they are, so I released the GIL

Reply:
Yes, better to use `CallState_begin` because it acquires the mutex `rk->rk_conf.sasl.lock` and other locks in `rd_kafka_all_brokers_wakeup` that are blocking operations

Reply:
Great, merging this now since there's agreement to release the GIL